src/app/utils.py

```diff
@@ -171,7 +173,153 @@ def makePullRequest(username, token, branchName, updateUpstream, fileName, commi
+        }
+    data = json.loads(response.text)
+    sha = str(data["object"]["sha"])
+
+    """ Getting names of all branches """
+    branch_url = url + "repos/%s/"+settings.NAMESPACE_REPO_NAME+"/branches"%(username)
+    response = requests.get(branch_url, headers=headers)
+    if response.status_code != 200:
+        logger.error("[Pull Request] Error occured while getting branch names, for %s user. "%(username)+response.text)
+        return {
+            "type":"error",
+            "message":"Some error occured while getting branch names. Please try again later or contact the SPDX Team."
+        }
+    data = json.loads(response.text)
+    branch_names = [i["name"] for i in data]
+
+    """ Creating branch """
+    if branchName in branch_names:
+        count=1
+        while True:
+            if((branchName+str(count)) in branch_names):
+                count+=1
+            else:
+                branchName = branchName+str(count)
+                break
+    create_branch_url = url + "repos/%s/"+settings.NAMESPACE_REPO_NAME+"/git/refs"%(username)
+    body = {
+        "ref":"refs/heads/"+branchName,
+        "sha":sha,
+    }
+    response = requests.post(create_branch_url, headers=headers, data=json.dumps(body))
+    if response.status_code != 201:
+        logger.error("[Pull Request] Error occured while creating branch, for %s user. "%(username)+response.text)
+        return {
+            "type":"error",
+            "message":"Some error occured while creating the branch. Please try again later or contact the SPDX Team."
+        }
+    data = json.loads(response.text)
+    branch_sha = data["object"]["sha"]
+
+    """ Creating Commit """
+    if fileName[-4:] == ".xml":
+        fileName = fileName[:-4]
+    fileName += ".xml"
+    commit_url = url + "repos/%s/"+settings.NAMESPACE_REPO_NAME+"/contents/src/%s"%(username, fileName)
+    xmlText = xmlText.encode('utf-8')
+    fileContent = base64.b64encode(xmlText)
+    body = {
+        "path":"src/"+fileName,
+        "message":commitMessage,
+        "content":fileContent,
+        "branch":branchName,
+    }
+    """ Check if file already exists """
+    file_url = "{0}//contents/src/{1}".format(TYPE_TO_URL_NAMESPACE[NORMAL], fileName)
+    response = requests.get(file_url, headers=headers)
+    if response.status_code == 200:
+        """ Creating Commit by updating the file """
+        data = json.loads(response.text)
+        file_sha = data["sha"]
+        body["sha"] = file_sha
+    response = requests.put(commit_url, headers=headers, data=json.dumps(body))
+    if not (response.status_code==201 or response.status_code==200):
+        logger.error("[Pull Request] Error occured while making commit, for %s user. "%(username)+response.text)
+        return {
+            "type":"error",
+            "message":"Some error occured while making commit. Please try again later or contact the SPDX Team."
```

Comment:
Could you use the same quoting style consistently? you are using single or double quotes at times. Use single quotes everywhere except in functions/objects docstrings.

Reply:
@pombredanne Ok. I will correct this.